test/unittest/tests/block/test_block_file.cpp

```diff
@@ -0,0 +1,214 @@
+{
+    WT_IGNORE_RET(__wti_bm_close_block(session, block));
+}
+
+void
+validate_block_fh(WT_BLOCK *block, uint expected_ref, std::string name)
+{
+    REQUIRE(block->fh != nullptr);
+    REQUIRE(std::string(block->fh->name).compare(name) == 0);
+    REQUIRE(block->fh->file_type == WT_FS_OPEN_FILE_TYPE_DATA);
+    REQUIRE(block->fh->ref == 1);
+}
+
+void
+validate_block_config(WT_BLOCK *block, std::map<std::string, std::string> &config_map)
+{
+    auto it = config_map.find("allocation_size");
+    uint32_t expected_alloc_size =
+      it != config_map.end() ? std::stoi(it->second) : std::stoi(ALLOCATION_SIZE);
+    REQUIRE(block->allocsize == expected_alloc_size);
+  
+    it = config_map.find("block_allocation");
+    uint32_t expected_block_allocation =
+      it != config_map.end() && it->second.compare(BLOCK_ALLOCATION) == 0 ? 0 : 1;
+  
+    REQUIRE(block->allocfirst == expected_block_allocation);
+    REQUIRE(block->os_cache_max == std::stoi(config_map["os_cache_max"]));
+    REQUIRE(block->os_cache_dirty_max == std::stoi(config_map["os_cache_dirty_max"]));
+}
+
+void
+validate_block(WT_BLOCK *block, std::map<std::string, std::string> &config_map, uint expected_ref,
+  std::string name, bool readonly = false, bool created_during_backup = false)
+{
+    REQUIRE(block != nullptr);
+
+    // Test Block immediate members.
+    REQUIRE(std::string(block->name).compare(name) == 0);
+    REQUIRE(block->objectid == WT_TIERED_OBJECTID_NONE);
+    // REQUIRE(block->compact_session_id = WT_SESSION_ID_INVALID);
+    REQUIRE(block->ref == expected_ref);
+    REQUIRE(block->readonly == readonly);
+    REQUIRE(block->created_during_backup == created_during_backup);
+    REQUIRE(block->extend_len == 0);
+
+    // Test Block file handle members.
+    validate_block_fh(block, expected_ref, name);
+    REQUIRE(std::string(block->live_lock.name).compare("block manager") == 0);
+    REQUIRE(block->live_lock.initialized == true);
+
+    // Test Block configuration members.
+    validate_block_config(block, config_map);
+}
+
+void
+validate_and_free_block(WT_SESSION_IMPL *session, WT_BLOCK *block, config_parser &cp,
+  uint expected_ref, std::string name, bool readonly = false, bool created_during_backup = false)
+{
+    validate_block(block, cp.get_config_map(), expected_ref, name, readonly, created_during_backup);
+    free_block(session, block);
+}
+
+TEST_CASE("Block: __wt_block_open", "[block_file]")
+{
+    /* Build Mock session, this will automatically create a mock connection. */
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    config_parser cp({{"allocation_size", ALLOCATION_SIZE}, {"block_allocation", BLOCK_ALLOCATION},
+      {"os_cache_max", OS_CACHE_MAX}, {"os_cache_dirty_max", OS_CACHE_DIRTY_MAX},
+      {"access_pattern_hint", ACCESS_PATTERN}});
+
+    REQUIRE((session->getMockConnection()->setupBlockManager(session->getWtSessionImpl())) == 0);
+
+    SECTION("Normal case")
+    {
+        WT_BLOCK *block;
```

Comment:
I suggest initialising all the `WT_BLOCK` pointers in the tests to `nullptr`, just in case.